packages/klighd-core/src/filtering/semantic-filtering-util.ts

```diff
@@ -0,0 +1,318 @@
+/*
+ * KIELER - Kiel Integrated Environment for Layout Eclipse RichClient
+ *
+ * http://rtsys.informatik.uni-kiel.de/kieler
+ *
+ * Copyright 2022 by
+ * + Kiel University
+ *   + Department of Computer Science
+ *     + Real-Time and Embedded Systems Group
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+import { SKGraphElement } from "../skgraph-models"
+
+//// Base constructs ////
+
+/**
+ * Base interface for semantic filter rules.
+ */
+export interface SemanticFilterRule {
+    /** The rule name is used to identify rules and distinguish them from one another. */
+    ruleName?: string
+    /** The default value is used to indicate whether the semantic filter should be on or off by default. */
+    defaultValue?: boolean
+}
+
+/**
+ * A semantic filter tag is used as a filter rule that evaluates to true iff the tag is present
+ * on a graph element.
+ */
+export class SemanticFilterTag implements SemanticFilterRule {
+    ruleName?: string
+    tag: string
+    /** Not defining num defaults it to 0. */
```

Comment:
When creating such an element here would default to it being `undefined` and not `0`. I guess here you refer to tags from the server will default to `0`? Then I would remove the `?` here as a value is guaranteed.